src/aws-ec2/instance-connect-endpoint.ts

```diff
@@ -0,0 +1,109 @@
+import { aws_ec2 } from 'aws-cdk-lib';
+import { Construct, IConstruct } from 'constructs';
+
+/**
+ * An EC2 Instance Connect Endpoint.
+ */
+export interface IInstanceConnectEndpoint extends aws_ec2.IConnectable, IConstruct {
+  /**
+   * The ID of the EC2 Instance Connect Endpoint.
+   *
+   * @attribute
+   */
+  readonly instanceConnectEndpointId: string;
+}
+
+/**
+ * Properties for defining an EC2 Instance Connect Endpoint.
+ */
+export interface InstanceConnectEndpointProps {
+  /**
+   * Unique, case-sensitive identifier that you provide to ensure the idempotency of the request.
+   *
+   * @see https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-instanceconnectendpoint.html#cfn-ec2-instanceconnectendpoint-clienttoken
+   */
+  readonly clientToken?: string;
```

Comment:
Is this really a useful property or bad API design inside the CloudFormation L1?

Reply:
This token is configurable to ensure idempotency in endpoint creation requests. The purpose is to ensure that resource creation requests remain idempotent during stack updates or redeployments, preventing unintended duplication of resources.

However, when using CloudFormation, failures in creation are rolled back, and it's not possible to create multiple endpoints in the same VPC, so there's no chance of duplication. Personally, I have never used this parameter and I consider it somewhat superfluous.

Therefore, I would not be opposed to removing it from the L2 arguments.

Reply:
I assume this is just BAD design from CloudFormation as this should be in the responsibility of CloudFormation

Reply:
Should this be removed from the L2 arguments?

Reply:
Let's leave it here.